static/js/stream_muting.js

```diff
@@ -67,4 +68,24 @@ export function update_is_muted(sub, value) {
         )}'] #sub_is_muted_setting .sub_setting_control`,
     );
     is_muted_checkbox.prop("checked", value);
+
+    $(".mute-note").toggleClass("hide-mute-note", !value);
+    $(".sub_notification_setting").toggleClass("muted-sub", value);
+    $(".sub_notification_setting").find("input[type='checkbox']").prop("disabled", value);
+
+    const row = $(`.stream-row[data-stream-id='${CSS.escape(sub.stream_id)}']`);
+    update_ui_muted_stream_checkboxes(row, value);
+}
+
+export function update_ui_muted_stream_checkboxes(row, value) {
+    $(row).toggleClass("control-label-disabled", value);
+    if (value) {
+        $(row).find(".unmute_stream").show();
+    } else {
+        $(row).find(".unmute_stream").hide();
+    }
+    $(row).find("input").prop("disabled", value);
+    $(row)
+        .find('[name="push_notifications"]')
+        .prop("disabled", !page_params.realm_push_notifications_enabled);
```

Comment:
This shoud also consider the `value` variable along with realm setting otherwise the checkbox remains enabled during live update if stream is muted but realm setting is set to True.